<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 8 Beta diversity | Introduction to microbiome data science</title>
  <meta name="description" content="Course material" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 8 Beta diversity | Introduction to microbiome data science" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Course material" />
  <meta name="github-repo" content="microbiome/course_2021_radboud" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 8 Beta diversity | Introduction to microbiome data science" />
  
  <meta name="twitter:description" content="Course material" />
  

<meta name="author" content="University of Turku" />


<meta name="date" content="2021-07-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="alpha-diversity.html"/>
<link rel="next" href="differential-abundance-analysis.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Introduction to microbiome data science</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Overview</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#introduction"><i class="fa fa-check"></i><b>1.1</b> Introduction</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#learning-goals"><i class="fa fa-check"></i><b>1.2</b> Learning goals</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#acknowledgments"><i class="fa fa-check"></i><b>1.3</b> Acknowledgments</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="program.html"><a href="program.html"><i class="fa fa-check"></i><b>2</b> Program</a>
<ul>
<li class="chapter" data-level="2.1" data-path="program.html"><a href="program.html#monday-12-july-from-raw-sequences-to-ecological-data-analysis"><i class="fa fa-check"></i><b>2.1</b> Monday 12 July: from raw sequences to ecological data analysis</a></li>
<li class="chapter" data-level="2.2" data-path="program.html"><a href="program.html#tuesday-13-july---alpha-diversity"><i class="fa fa-check"></i><b>2.2</b> Tuesday 13 July - Alpha diversity</a></li>
<li class="chapter" data-level="2.3" data-path="program.html"><a href="program.html#wednesday-14-july---beta-diversity"><i class="fa fa-check"></i><b>2.3</b> Wednesday 14 July - Beta diversity</a></li>
<li class="chapter" data-level="2.4" data-path="program.html"><a href="program.html#thursday-15-july---differential-abundance"><i class="fa fa-check"></i><b>2.4</b> Thursday 15 July - Differential abundance</a></li>
<li class="chapter" data-level="2.5" data-path="program.html"><a href="program.html#friday-16-july-presentations-closing"><i class="fa fa-check"></i><b>2.5</b> Friday 16 July: Presentations &amp; closing</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getting-started.html"><a href="getting-started.html"><i class="fa fa-check"></i><b>3</b> Getting started</a>
<ul>
<li class="chapter" data-level="3.1" data-path="getting-started.html"><a href="getting-started.html#checklist-before-the-course"><i class="fa fa-check"></i><b>3.1</b> Checklist (before the course)</a></li>
<li class="chapter" data-level="3.2" data-path="getting-started.html"><a href="getting-started.html#support-and-resources"><i class="fa fa-check"></i><b>3.2</b> Support and resources</a></li>
<li class="chapter" data-level="3.3" data-path="getting-started.html"><a href="getting-started.html#installing-and-loading-the-required-r-packages"><i class="fa fa-check"></i><b>3.3</b> Installing and loading the required R packages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="reproducible-reporting-with-rmarkdown.html"><a href="reproducible-reporting-with-rmarkdown.html"><i class="fa fa-check"></i><b>4</b> Reproducible reporting with Rmarkdown</a></li>
<li class="chapter" data-level="5" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html"><i class="fa fa-check"></i><b>5</b> Importing microbiome data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#data-access"><i class="fa fa-check"></i><b>5.1</b> Data access</a></li>
<li class="chapter" data-level="5.2" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#importing-microbiome-data-in-r"><i class="fa fa-check"></i><b>5.2</b> Importing microbiome data in R</a></li>
<li class="chapter" data-level="5.3" data-path="importing-microbiome-data.html"><a href="importing-microbiome-data.html#example-solutions"><i class="fa fa-check"></i><b>5.3</b> Example solutions</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html"><i class="fa fa-check"></i><b>6</b> Microbiome data exploration</a>
<ul>
<li class="chapter" data-level="6.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#data-structure"><i class="fa fa-check"></i><b>6.1</b> Data structure</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#transformations"><i class="fa fa-check"></i><b>6.1.1</b> Transformations</a></li>
<li class="chapter" data-level="6.1.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#aggregation"><i class="fa fa-check"></i><b>6.1.2</b> Aggregation</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#visualization"><i class="fa fa-check"></i><b>6.2</b> Visualization</a></li>
<li class="chapter" data-level="6.3" data-path="microbiome-data-exploration.html"><a href="microbiome-data-exploration.html#exercises-optional"><i class="fa fa-check"></i><b>6.3</b> Exercises (optional)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="alpha-diversity.html"><a href="alpha-diversity.html"><i class="fa fa-check"></i><b>7</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="7.1" data-path="alpha-diversity.html"><a href="alpha-diversity.html#visualization-1"><i class="fa fa-check"></i><b>7.1</b> Visualization</a></li>
<li class="chapter" data-level="7.2" data-path="alpha-diversity.html"><a href="alpha-diversity.html#statistical-testing-and-comparisons"><i class="fa fa-check"></i><b>7.2</b> Statistical testing and comparisons</a></li>
<li class="chapter" data-level="7.3" data-path="alpha-diversity.html"><a href="alpha-diversity.html#exercises"><i class="fa fa-check"></i><b>7.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="beta-diversity.html"><a href="beta-diversity.html"><i class="fa fa-check"></i><b>8</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="8.1" data-path="beta-diversity.html"><a href="beta-diversity.html#examples-of-pcoa-with-different-settings"><i class="fa fa-check"></i><b>8.1</b> Examples of PCoA with different settings</a>
<ul>
<li class="chapter" data-level="8.1.1" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-bray-curtis"><i class="fa fa-check"></i><b>8.1.1</b> PCoA for ASV-level data with Bray-Curtis</a></li>
<li class="chapter" data-level="8.1.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-for-asv-level-data-with-aitchison-distance"><i class="fa fa-check"></i><b>8.1.2</b> PCoA for ASV-level data with Aitchison distance</a></li>
<li class="chapter" data-level="8.1.3" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-aggregated-to-phylum-level"><i class="fa fa-check"></i><b>8.1.3</b> PCoA aggregated to Phylum level</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="beta-diversity.html"><a href="beta-diversity.html#highlighting-external-variables"><i class="fa fa-check"></i><b>8.2</b> Highlighting external variables</a>
<ul>
<li class="chapter" data-level="8.2.1" data-path="beta-diversity.html"><a href="beta-diversity.html#discrete-grouping-variable-shown-with-colors"><i class="fa fa-check"></i><b>8.2.1</b> Discrete grouping variable shown with colors</a></li>
<li class="chapter" data-level="8.2.2" data-path="beta-diversity.html"><a href="beta-diversity.html#pcoa-plot-with-continuous-variable"><i class="fa fa-check"></i><b>8.2.2</b> PCoA plot with continuous variable</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="beta-diversity.html"><a href="beta-diversity.html#estimating-associations-with-an-external-variable"><i class="fa fa-check"></i><b>8.3</b> Estimating associations with an external variable</a></li>
<li class="chapter" data-level="8.4" data-path="beta-diversity.html"><a href="beta-diversity.html#community-typing"><i class="fa fa-check"></i><b>8.4</b> Community typing</a></li>
<li class="chapter" data-level="8.5" data-path="beta-diversity.html"><a href="beta-diversity.html#exercises-1"><i class="fa fa-check"></i><b>8.5</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html"><i class="fa fa-check"></i><b>9</b> Differential abundance analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#wilcoxon-test"><i class="fa fa-check"></i><b>9.1</b> Wilcoxon test</a></li>
<li class="chapter" data-level="9.2" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#deseq2"><i class="fa fa-check"></i><b>9.2</b> DESeq2</a></li>
<li class="chapter" data-level="9.3" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#ancom-bc"><i class="fa fa-check"></i><b>9.3</b> ANCOM-BC</a></li>
<li class="chapter" data-level="9.4" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-the-methods"><i class="fa fa-check"></i><b>9.4</b> Comparison of the methods</a></li>
<li class="chapter" data-level="9.5" data-path="differential-abundance-analysis.html"><a href="differential-abundance-analysis.html#comparison-of-abundance"><i class="fa fa-check"></i><b>9.5</b> Comparison of abundance</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="study-material.html"><a href="study-material.html"><i class="fa fa-check"></i><b>10</b> Study material</a>
<ul>
<li class="chapter" data-level="10.1" data-path="study-material.html"><a href="study-material.html#lecture-slides"><i class="fa fa-check"></i><b>10.1</b> Lecture slides</a></li>
<li class="chapter" data-level="10.2" data-path="study-material.html"><a href="study-material.html#example-solutions-1"><i class="fa fa-check"></i><b>10.2</b> Example solutions</a></li>
<li class="chapter" data-level="10.3" data-path="study-material.html"><a href="study-material.html#r-programming-resources"><i class="fa fa-check"></i><b>10.3</b> R programming resources</a></li>
<li class="chapter" data-level="10.4" data-path="study-material.html"><a href="study-material.html#resources-for-treesummarizedexperiment"><i class="fa fa-check"></i><b>10.4</b> Resources for TreeSummarizedExperiment</a></li>
<li class="chapter" data-level="10.5" data-path="study-material.html"><a href="study-material.html#resources-for-phyloseq"><i class="fa fa-check"></i><b>10.5</b> Resources for phyloseq</a></li>
<li class="chapter" data-level="10.6" data-path="study-material.html"><a href="study-material.html#further-reading"><i class="fa fa-check"></i><b>10.6</b> Further reading</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html"><i class="fa fa-check"></i><b>11</b> Miscellaneous material</a>
<ul>
<li class="chapter" data-level="11.1" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#shapiro-wilk-test"><i class="fa fa-check"></i><b>11.1</b> Shapiro-Wilk test</a></li>
<li class="chapter" data-level="11.2" data-path="miscellaneous-material.html"><a href="miscellaneous-material.html#deseq-details"><i class="fa fa-check"></i><b>11.2</b> Deseq details</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="exercise-solutions.html"><a href="exercise-solutions.html"><i class="fa fa-check"></i><b>12</b> Exercise Solutions</a>
<ul>
<li class="chapter" data-level="12.1" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-5"><i class="fa fa-check"></i><b>12.1</b> Section 5</a></li>
<li class="chapter" data-level="12.2" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-6"><i class="fa fa-check"></i><b>12.2</b> Section 6</a></li>
<li class="chapter" data-level="12.3" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-7"><i class="fa fa-check"></i><b>12.3</b> Section 7</a></li>
<li class="chapter" data-level="12.4" data-path="exercise-solutions.html"><a href="exercise-solutions.html#section-8"><i class="fa fa-check"></i><b>12.4</b> Section 8</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Introduction to microbiome data science</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="beta-diversity" class="section level1" number="8">
<h1><span class="header-section-number">Chapter 8</span> Beta diversity</h1>
<p>Beta diversity is another name for sample dissimilarity. It quantifies
differences in the overall taxonomic composition between two samples.</p>
<p>Common indices include Bray-Curtis, Unifrac, Jaccard index, and the
Aitchison distance. Each of these (dis)similarity measures emphasizes
different aspects. For example, UniFrac incorporates phylogenetic
information, and Jaccard index ignores exact abundances and considers
only presence/absence values. For more background information
and examples, you can check the dedicated section in <a href="https://microbiome.github.io/OMA/microbiome-diversity.html#beta-diversity">online
book</a>.</p>
<div id="examples-of-pcoa-with-different-settings" class="section level2" number="8.1">
<h2><span class="header-section-number">8.1</span> Examples of PCoA with different settings</h2>
<p>Beta diversity estimation generates a (dis)similarity matrix that
contains for each sample (rows) the dissimilarity to any other sample
(columns).</p>
<p>This complex set of pairwise relations can be visualized in
informative ways, and even coupled with other explanatory
variables. As a first step, we compress the information to a lower
dimensionality, or fewer principal components, and then visualize
sample similarity based on that using ordination techniques, such as
Principal Coordinate Analysis (PCoA). PCoA is a non-linear dimension
reduction technique, and with Euclidean distances it is is identical
to the linear PCA (except for potential scaling).</p>
<p>We typically retain just the two (or three) most informative top
components, and ignore the other information. Each sample has a score
on each of these components, and each component measures the variation
across a set of correlated taxa. The top components are then easily
visualized on a two (or three) dimensional display.</p>
<p>Let us next look at some concrete examples.</p>
<div id="pcoa-for-asv-level-data-with-bray-curtis" class="section level3" number="8.1.1">
<h3><span class="header-section-number">8.1.1</span> PCoA for ASV-level data with Bray-Curtis</h3>
<p>Let us start with PCoA based on a Bray-Curtis dissimilarity matrix
calculated at Genus level abundances.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="beta-diversity.html#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick the relative abundance table</span></span>
<span id="cb145-2"><a href="beta-diversity.html#cb145-2" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>relabundance</span>
<span id="cb145-3"><a href="beta-diversity.html#cb145-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-4"><a href="beta-diversity.html#cb145-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Bray-Curtis distances between samples. Because taxa is in</span></span>
<span id="cb145-5"><a href="beta-diversity.html#cb145-5" aria-hidden="true" tabindex="-1"></a><span class="co"># columns, it is used to compare different samples. We transpose the</span></span>
<span id="cb145-6"><a href="beta-diversity.html#cb145-6" aria-hidden="true" tabindex="-1"></a><span class="co"># assay to get taxa to columns</span></span>
<span id="cb145-7"><a href="beta-diversity.html#cb145-7" aria-hidden="true" tabindex="-1"></a>bray_curtis_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(<span class="fu">t</span>(rel_abund_assay), <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb145-8"><a href="beta-diversity.html#cb145-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-9"><a href="beta-diversity.html#cb145-9" aria-hidden="true" tabindex="-1"></a><span class="co"># PCoA</span></span>
<span id="cb145-10"><a href="beta-diversity.html#cb145-10" aria-hidden="true" tabindex="-1"></a>bray_curtis_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(bray_curtis_dist)</span>
<span id="cb145-11"><a href="beta-diversity.html#cb145-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-12"><a href="beta-diversity.html#cb145-12" aria-hidden="true" tabindex="-1"></a><span class="co"># All components could be found here: </span></span>
<span id="cb145-13"><a href="beta-diversity.html#cb145-13" aria-hidden="true" tabindex="-1"></a><span class="co"># bray_curtis_pcoa$vectors</span></span>
<span id="cb145-14"><a href="beta-diversity.html#cb145-14" aria-hidden="true" tabindex="-1"></a><span class="co"># But we only need the first two to demonstrate what we can do:</span></span>
<span id="cb145-15"><a href="beta-diversity.html#cb145-15" aria-hidden="true" tabindex="-1"></a>bray_curtis_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> bray_curtis_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb145-16"><a href="beta-diversity.html#cb145-16" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">pcoa2 =</span> bray_curtis_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb145-17"><a href="beta-diversity.html#cb145-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-18"><a href="beta-diversity.html#cb145-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a plot</span></span>
<span id="cb145-19"><a href="beta-diversity.html#cb145-19" aria-hidden="true" tabindex="-1"></a>bray_curtis_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> bray_curtis_pcoa_df, <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb145-20"><a href="beta-diversity.html#cb145-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb145-21"><a href="beta-diversity.html#cb145-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>,</span>
<span id="cb145-22"><a href="beta-diversity.html#cb145-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>, </span>
<span id="cb145-23"><a href="beta-diversity.html#cb145-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Bray-Curtis PCoA&quot;</span>) <span class="sc">+</span></span>
<span id="cb145-24"><a href="beta-diversity.html#cb145-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">10</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb145-25"><a href="beta-diversity.html#cb145-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb145-26"><a href="beta-diversity.html#cb145-26" aria-hidden="true" tabindex="-1"></a>bray_curtis_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/pcoa_asv_bc-1.png" width="672" /></p>
</div>
<div id="pcoa-for-asv-level-data-with-aitchison-distance" class="section level3" number="8.1.2">
<h3><span class="header-section-number">8.1.2</span> PCoA for ASV-level data with Aitchison distance</h3>
<p>Now the same using Aitchison distance. This metric corresponds to
Euclidean distances between CLR transformed sample abundance vectors.</p>
<div class="sourceCode" id="cb146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb146-1"><a href="beta-diversity.html#cb146-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Pseudocount is added, because data contains zeros. </span></span>
<span id="cb146-2"><a href="beta-diversity.html#cb146-2" aria-hidden="true" tabindex="-1"></a>tse <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb146-3"><a href="beta-diversity.html#cb146-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-4"><a href="beta-diversity.html#cb146-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb146-5"><a href="beta-diversity.html#cb146-5" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>clr</span>
<span id="cb146-6"><a href="beta-diversity.html#cb146-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-7"><a href="beta-diversity.html#cb146-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb146-8"><a href="beta-diversity.html#cb146-8" aria-hidden="true" tabindex="-1"></a>clr_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_assay)</span>
<span id="cb146-9"><a href="beta-diversity.html#cb146-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-10"><a href="beta-diversity.html#cb146-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb146-11"><a href="beta-diversity.html#cb146-11" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb146-12"><a href="beta-diversity.html#cb146-12" aria-hidden="true" tabindex="-1"></a>euclidean_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb146-13"><a href="beta-diversity.html#cb146-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-14"><a href="beta-diversity.html#cb146-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb146-15"><a href="beta-diversity.html#cb146-15" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_dist)</span>
<span id="cb146-16"><a href="beta-diversity.html#cb146-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-17"><a href="beta-diversity.html#cb146-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb146-18"><a href="beta-diversity.html#cb146-18" aria-hidden="true" tabindex="-1"></a>euclidean_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">pcoa1 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb146-19"><a href="beta-diversity.html#cb146-19" aria-hidden="true" tabindex="-1"></a>                                <span class="at">pcoa2 =</span> euclidean_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb146-20"><a href="beta-diversity.html#cb146-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-21"><a href="beta-diversity.html#cb146-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates the plot</span></span>
<span id="cb146-22"><a href="beta-diversity.html#cb146-22" aria-hidden="true" tabindex="-1"></a>euclidean_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_pcoa_df, <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb146-23"><a href="beta-diversity.html#cb146-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb146-24"><a href="beta-diversity.html#cb146-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>,</span>
<span id="cb146-25"><a href="beta-diversity.html#cb146-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>,</span>
<span id="cb146-26"><a href="beta-diversity.html#cb146-26" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Euclidean PCoA with CLR transformation&quot;</span>) <span class="sc">+</span></span>
<span id="cb146-27"><a href="beta-diversity.html#cb146-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb146-28"><a href="beta-diversity.html#cb146-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb146-29"><a href="beta-diversity.html#cb146-29" aria-hidden="true" tabindex="-1"></a>euclidean_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/pcoa_asv_aitchison-1.png" width="672" /></p>
</div>
<div id="pcoa-aggregated-to-phylum-level" class="section level3" number="8.1.3">
<h3><span class="header-section-number">8.1.3</span> PCoA aggregated to Phylum level</h3>
<p>We use again the Aitchison distances in this example but this time applied to the phylum level.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="beta-diversity.html#cb147-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Does clr transformation. Psuedocount is added, because data contains zeros. </span></span>
<span id="cb147-2"><a href="beta-diversity.html#cb147-2" aria-hidden="true" tabindex="-1"></a>tse_phylum <span class="ot">&lt;-</span> <span class="fu">transformCounts</span>(tse_phylum, <span class="at">method =</span> <span class="st">&quot;clr&quot;</span>, <span class="at">pseudocount =</span> <span class="dv">1</span>)</span>
<span id="cb147-3"><a href="beta-diversity.html#cb147-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-4"><a href="beta-diversity.html#cb147-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets clr table</span></span>
<span id="cb147-5"><a href="beta-diversity.html#cb147-5" aria-hidden="true" tabindex="-1"></a>clr_phylum_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse_phylum)<span class="sc">$</span>clr</span>
<span id="cb147-6"><a href="beta-diversity.html#cb147-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-7"><a href="beta-diversity.html#cb147-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Transposes it to get taxa to columns</span></span>
<span id="cb147-8"><a href="beta-diversity.html#cb147-8" aria-hidden="true" tabindex="-1"></a>clr_phylum_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(clr_phylum_assay)</span>
<span id="cb147-9"><a href="beta-diversity.html#cb147-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-10"><a href="beta-diversity.html#cb147-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates Euclidean distances between samples. Because taxa is in columns,</span></span>
<span id="cb147-11"><a href="beta-diversity.html#cb147-11" aria-hidden="true" tabindex="-1"></a><span class="co"># it is used to compare different samples.</span></span>
<span id="cb147-12"><a href="beta-diversity.html#cb147-12" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_dist <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">vegdist</span>(clr_assay, <span class="at">method =</span> <span class="st">&quot;euclidean&quot;</span>)</span>
<span id="cb147-13"><a href="beta-diversity.html#cb147-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-14"><a href="beta-diversity.html#cb147-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Does principal coordinate analysis</span></span>
<span id="cb147-15"><a href="beta-diversity.html#cb147-15" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_pcoa <span class="ot">&lt;-</span> ecodist<span class="sc">::</span><span class="fu">pco</span>(euclidean_phylum_dist)</span>
<span id="cb147-16"><a href="beta-diversity.html#cb147-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-17"><a href="beta-diversity.html#cb147-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a data frame from principal coordinates</span></span>
<span id="cb147-18"><a href="beta-diversity.html#cb147-18" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_pcoa_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb147-19"><a href="beta-diversity.html#cb147-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">pcoa1 =</span> euclidean_phylum_pcoa<span class="sc">$</span>vectors[,<span class="dv">1</span>], </span>
<span id="cb147-20"><a href="beta-diversity.html#cb147-20" aria-hidden="true" tabindex="-1"></a>  <span class="at">pcoa2 =</span> euclidean_phylum_pcoa<span class="sc">$</span>vectors[,<span class="dv">2</span>])</span>
<span id="cb147-21"><a href="beta-diversity.html#cb147-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-22"><a href="beta-diversity.html#cb147-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb147-23"><a href="beta-diversity.html#cb147-23" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_phylum_pcoa_df,</span>
<span id="cb147-24"><a href="beta-diversity.html#cb147-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2)) <span class="sc">+</span></span>
<span id="cb147-25"><a href="beta-diversity.html#cb147-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb147-26"><a href="beta-diversity.html#cb147-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>,</span>
<span id="cb147-27"><a href="beta-diversity.html#cb147-27" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>,</span>
<span id="cb147-28"><a href="beta-diversity.html#cb147-28" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;Aitchison distances at Phylum level&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb147-29"><a href="beta-diversity.html#cb147-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb147-30"><a href="beta-diversity.html#cb147-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb147-31"><a href="beta-diversity.html#cb147-31" aria-hidden="true" tabindex="-1"></a>euclidean_phylum_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/pcoa_phylum_aitchison-1.png" width="672" /></p>
</div>
</div>
<div id="highlighting-external-variables" class="section level2" number="8.2">
<h2><span class="header-section-number">8.2</span> Highlighting external variables</h2>
<p>We can map other variables on the same plot for example by coloring
the points accordingly.</p>
<div id="discrete-grouping-variable-shown-with-colors" class="section level3" number="8.2.1">
<h3><span class="header-section-number">8.2.1</span> Discrete grouping variable shown with colors</h3>
<div class="sourceCode" id="cb148"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb148-1"><a href="beta-diversity.html#cb148-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds the variable we later use for coloring to the data frame</span></span>
<span id="cb148-2"><a href="beta-diversity.html#cb148-2" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb148-3"><a href="beta-diversity.html#cb148-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">patient_status =</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>patient_status)</span>
<span id="cb148-4"><a href="beta-diversity.html#cb148-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-5"><a href="beta-diversity.html#cb148-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb148-6"><a href="beta-diversity.html#cb148-6" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_patient_status_pcoa_df, </span>
<span id="cb148-7"><a href="beta-diversity.html#cb148-7" aria-hidden="true" tabindex="-1"></a>                                        <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb148-8"><a href="beta-diversity.html#cb148-8" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">color =</span> patient_status)) <span class="sc">+</span></span>
<span id="cb148-9"><a href="beta-diversity.html#cb148-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb148-10"><a href="beta-diversity.html#cb148-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>,</span>
<span id="cb148-11"><a href="beta-diversity.html#cb148-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>,</span>
<span id="cb148-12"><a href="beta-diversity.html#cb148-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span></span>
<span id="cb148-13"><a href="beta-diversity.html#cb148-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb148-14"><a href="beta-diversity.html#cb148-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb148-15"><a href="beta-diversity.html#cb148-15" aria-hidden="true" tabindex="-1"></a>euclidean_patient_status_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/pcoa_genus-1.png" width="672" /></p>
</div>
<div id="pcoa-plot-with-continuous-variable" class="section level3" number="8.2.2">
<h3><span class="header-section-number">8.2.2</span> PCoA plot with continuous variable</h3>
<p>We can do the same as above using any continuous variable. E.g. let us
see how the plotted samples differ in their alpha diversity (as an
example of a continuous variable):</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="beta-diversity.html#cb149-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds coloring information to the data frame, creates new column</span></span>
<span id="cb149-2"><a href="beta-diversity.html#cb149-2" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_pcoa_df <span class="ot">&lt;-</span> <span class="fu">cbind</span>(euclidean_pcoa_df,</span>
<span id="cb149-3"><a href="beta-diversity.html#cb149-3" aria-hidden="true" tabindex="-1"></a>                             <span class="at">shannon =</span> <span class="fu">colData</span>(tse)<span class="sc">$</span>Shannon_diversity)</span>
<span id="cb149-4"><a href="beta-diversity.html#cb149-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-5"><a href="beta-diversity.html#cb149-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creates a plot</span></span>
<span id="cb149-6"><a href="beta-diversity.html#cb149-6" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data =</span> euclidean_shannon_pcoa_df, </span>
<span id="cb149-7"><a href="beta-diversity.html#cb149-7" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">aes</span>(<span class="at">x=</span>pcoa1, <span class="at">y=</span>pcoa2,</span>
<span id="cb149-8"><a href="beta-diversity.html#cb149-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">color =</span> shannon)) <span class="sc">+</span> </span>
<span id="cb149-9"><a href="beta-diversity.html#cb149-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb149-10"><a href="beta-diversity.html#cb149-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;PC1&quot;</span>,</span>
<span id="cb149-11"><a href="beta-diversity.html#cb149-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">&quot;PC2&quot;</span>,</span>
<span id="cb149-12"><a href="beta-diversity.html#cb149-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">title =</span> <span class="st">&quot;PCoA with Aitchison distances&quot;</span>) <span class="sc">+</span></span>
<span id="cb149-13"><a href="beta-diversity.html#cb149-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>)) <span class="co"># makes titles smaller</span></span>
<span id="cb149-14"><a href="beta-diversity.html#cb149-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb149-15"><a href="beta-diversity.html#cb149-15" aria-hidden="true" tabindex="-1"></a>euclidean_shannon_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/pcoa_coloring-1.png" width="672" /></p>
</div>
</div>
<div id="estimating-associations-with-an-external-variable" class="section level2" number="8.3">
<h2><span class="header-section-number">8.3</span> Estimating associations with an external variable</h2>
<p>Next to visualizing whether any variable is associated with
differences between samples, we can also quantify the strength of the
association between community composition (beta diversity) and
external factors.</p>
<p>The standard way to do this is to perform a so-called permutational
multivariate analysis of variance (PERMANOVA). This method takes as
input the abundance table, which measure of distance you want to base
the test on and a formula that tells the model how you think the
variables are associated with each other.</p>
<div class="sourceCode" id="cb150"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb150-1"><a href="beta-diversity.html#cb150-1" aria-hidden="true" tabindex="-1"></a><span class="co"># First we get the relative abundance table</span></span>
<span id="cb150-2"><a href="beta-diversity.html#cb150-2" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">assays</span>(tse)<span class="sc">$</span>relabundance</span>
<span id="cb150-3"><a href="beta-diversity.html#cb150-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-4"><a href="beta-diversity.html#cb150-4" aria-hidden="true" tabindex="-1"></a><span class="co"># again transpose it to get taxa to columns</span></span>
<span id="cb150-5"><a href="beta-diversity.html#cb150-5" aria-hidden="true" tabindex="-1"></a>rel_abund_assay <span class="ot">&lt;-</span> <span class="fu">t</span>(rel_abund_assay)</span>
<span id="cb150-6"><a href="beta-diversity.html#cb150-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-7"><a href="beta-diversity.html#cb150-7" aria-hidden="true" tabindex="-1"></a><span class="co"># then we can perform the method</span></span>
<span id="cb150-8"><a href="beta-diversity.html#cb150-8" aria-hidden="true" tabindex="-1"></a>permanova_cohort <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis</span>(rel_abund_assay <span class="sc">~</span> cohort,</span>
<span id="cb150-9"><a href="beta-diversity.html#cb150-9" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">data =</span> <span class="fu">colData</span>(tse),</span>
<span id="cb150-10"><a href="beta-diversity.html#cb150-10" aria-hidden="true" tabindex="-1"></a>                                  <span class="at">permutations =</span> <span class="dv">9999</span>)</span>
<span id="cb150-11"><a href="beta-diversity.html#cb150-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb150-12"><a href="beta-diversity.html#cb150-12" aria-hidden="true" tabindex="-1"></a><span class="co"># we can obtain a the p value for our predictor:</span></span>
<span id="cb150-13"><a href="beta-diversity.html#cb150-13" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Different different cohorts and variance of abundance &quot;</span>,</span>
<span id="cb150-14"><a href="beta-diversity.html#cb150-14" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;between samples, p-value: &quot;</span>, </span>
<span id="cb150-15"><a href="beta-diversity.html#cb150-15" aria-hidden="true" tabindex="-1"></a>              <span class="fu">as.data.frame</span>(permanova_cohort<span class="sc">$</span>aov.tab)[<span class="st">&quot;cohort&quot;</span>, <span class="st">&quot;Pr(&gt;F)&quot;</span>]))</span></code></pre></div>
<pre><code>## [1] &quot;Different different cohorts and variance of abundance between samples, p-value: 0.7427&quot;</code></pre>
<p>The cohort variable is not significantly associated with
microbiota composition (p-value is over 0.05).</p>
<p>We can, however, visualize those taxa whose abundances drive the
differences between cohorts. We first need to extract the model
coefficients of taxa:</p>
<div class="sourceCode" id="cb152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb152-1"><a href="beta-diversity.html#cb152-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the coefficients</span></span>
<span id="cb152-2"><a href="beta-diversity.html#cb152-2" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">&lt;-</span> <span class="fu">coefficients</span>(permanova_cohort)[<span class="st">&quot;cohort1&quot;</span>,]</span>
<span id="cb152-3"><a href="beta-diversity.html#cb152-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-4"><a href="beta-diversity.html#cb152-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets the highest coefficients</span></span>
<span id="cb152-5"><a href="beta-diversity.html#cb152-5" aria-hidden="true" tabindex="-1"></a>top.coef <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">head</span>(coef[<span class="fu">rev</span>(<span class="fu">order</span>(<span class="fu">abs</span>(coef)))],<span class="dv">20</span>))</span>
<span id="cb152-6"><a href="beta-diversity.html#cb152-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-7"><a href="beta-diversity.html#cb152-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Plots the coefficients</span></span>
<span id="cb152-8"><a href="beta-diversity.html#cb152-8" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> top.coef,</span>
<span id="cb152-9"><a href="beta-diversity.html#cb152-9" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">y =</span> <span class="fu">factor</span>(<span class="fu">names</span>(top.coef),</span>
<span id="cb152-10"><a href="beta-diversity.html#cb152-10" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">unique</span>(<span class="fu">names</span>(top.coef)))),</span>
<span id="cb152-11"><a href="beta-diversity.html#cb152-11" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span></span>
<span id="cb152-12"><a href="beta-diversity.html#cb152-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>) <span class="sc">+</span></span>
<span id="cb152-13"><a href="beta-diversity.html#cb152-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;&quot;</span>, <span class="at">y=</span><span class="st">&quot;&quot;</span>, <span class="at">title=</span><span class="st">&quot;Top Taxa&quot;</span>) <span class="sc">+</span></span>
<span id="cb152-14"><a href="beta-diversity.html#cb152-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span>
<span id="cb152-15"><a href="beta-diversity.html#cb152-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb152-16"><a href="beta-diversity.html#cb152-16" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/permanova_coefs-1.png" width="672" /></p>
<p>The above plot shows taxa as code names, and it is hard to tell which
bacterial groups they represent. However, it is easy to add human readable
names. We can fetch those from our rowData. Here we use Genus level names:</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="beta-diversity.html#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Gets corresponding Genus level names and stores them to top.coef</span></span>
<span id="cb153-2"><a href="beta-diversity.html#cb153-2" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">rowData</span>(tse)[<span class="fu">names</span>(top.coef), ][,<span class="st">&quot;Genus&quot;</span>]</span>
<span id="cb153-3"><a href="beta-diversity.html#cb153-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb153-4"><a href="beta-diversity.html#cb153-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Adds new labels to the plot</span></span>
<span id="cb153-5"><a href="beta-diversity.html#cb153-5" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot <span class="ot">&lt;-</span> top_taxa_coeffient_plot <span class="sc">+</span></span>
<span id="cb153-6"><a href="beta-diversity.html#cb153-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> names) <span class="co"># Adds new labels</span></span>
<span id="cb153-7"><a href="beta-diversity.html#cb153-7" aria-hidden="true" tabindex="-1"></a>top_taxa_coeffient_plot</span></code></pre></div>
<p><img src="07-beta_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<p>There are many alternative and complementary methods for analysing
community composition. For more examples, see a dedicated section on
beta diversity in the <a href="https://microbiome.github.io/OMA/microbiome-diversity.html#beta-diversity">online
book</a>.</p>
</div>
<div id="community-typing" class="section level2" number="8.4">
<h2><span class="header-section-number">8.4</span> Community typing</h2>
<p>A dedicated section presenting examples on community typing is in the
<a href="https://microbiome.github.io/OMA/microbiome-community.html#community-typing">online book</a>.</p>
</div>
<div id="exercises-1" class="section level2" number="8.5">
<h2><span class="header-section-number">8.5</span> Exercises</h2>
<ul>
<li><p>Visualize community variation with different methods (PCA, MDS, t-SNE) by using the options in the alternative method, plotReducedDim <a href="https://microbiome.github.io/OMA/microbiome-diversity.html#estimating-beta-diversity">OMA</a>. Compare results obtained with different dissimilarities (Euclidean, Bray-Curtis, Unifrac..) and transformations (CLR, compositional..) of your own choice."</p></li>
<li><p>Investigate the influence of the data transformations on
statistical analysis: Visualize community variation with PCoA with
the following options: 1) Bray-Curtis distances for compositional
data; 2) Euclidean distances for CLR-transformed data.</p></li>
<li><p>Community-level comparisons: Use PERMANOVA to investigate whether
the community composition differs between two groups of individuals
(e.g.males and females, or some other grouping of your
choice). You can also include covariates such as age and gender,
and see how this affects the results.</p></li>
<li><p>Perform community typing for the data using the DMM method <a href="https://microbiome.github.io/OMA/microbiome-community.html#community-typing">OMA</a></p></li>
<li><p>Example <a href="08-5-ex-sol-ADHD.html">Solutions</a></p></li>
</ul>

<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## Loading required package: SingleCellExperiment</code></pre>
<pre><code>## Loading required package: TreeSummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: &#39;Biostrings&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: ggraph</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:Biostrings&#39;:
## 
##     collapse, intersect, setdiff, setequal, union</code></pre>
<pre><code>## The following object is masked from &#39;package:XVector&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:Biobase&#39;:
## 
##     combine</code></pre>
<pre><code>## The following objects are masked from &#39;package:GenomicRanges&#39;:
## 
##     intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:GenomeInfoDb&#39;:
## 
##     intersect</code></pre>
<pre><code>## The following objects are masked from &#39;package:IRanges&#39;:
## 
##     collapse, desc, intersect, setdiff, slice, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:S4Vectors&#39;:
## 
##     first, intersect, rename, setdiff, setequal, union</code></pre>
<pre><code>## The following objects are masked from &#39;package:BiocGenerics&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:matrixStats&#39;:
## 
##     count</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="alpha-diversity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-abundance-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"linkedin": true,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["radboud2021_material.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
