--- 
title: "Introduction to microbiome data science"
author: "`r authors`"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: microbiome/course_2021_radboud
description: "Course material"
output:
  bookdown::pdf_book
  bookdown::gitbook
always_allow_html: yes
classoption: oneside
geometry:
  - top=30mm
  - left=15mm
  
---

# Overview 

**Welcome to [Radboud Summer School, July 2021](https://www.ru.nl/radboudsummerschool/courses/2021/brain-bacteria-behaviour/)**

<img src="https://user-images.githubusercontent.com/60338854/121848694-1072a480-ccf3-11eb-9af2-7fdefd8d1794.png" alt="ML4microbiome" width="50%"/>

<p style="font-size:12px">Figure source: Moreno-Indias _et al_. (2021) [Statistical and Machine Learning Techniques in Human Microbiome Studies: Contemporary Challenges and Solutions](https://doi.org/10.3389/fmicb.2021.635781). Frontiers in Microbiology 12:11.</p>




## Introduction

This course is based on [_miaverse_](https://microbiome.github.io) (mia = **MI**crobiome **A**nalysis) is an
R/Bioconductor framework for microbiome data science. It aims to
extend the capabilities of another popular framework,
[phyloseq](https://joey711.github.io/phyloseq/).

The miaverse framework consists of an efficient data structure, an
associated package ecosystem, demonstration data sets, and open
documentation. These are explained in more detail in the online book
[Orchestrating Microbiome Analysis](https://microbiome.github.io/OMA).

This training material walks you through an example workflow that
shows the standard steps of taxonomic data analysis covering data
access, exploration, analysis, visualization and reporoducible
reporting. **You can run the workflow by simply copy-pasting the
examples.** For advanced material, you can test and modify further
examples from the [OMA book](https://microbiome.github.io/OMA), or try
to apply the techniques to your own data.




## Learning goals

This course provides an overview of the standard bioinformatics
workflow in taxonomic profiling studies, ranging from data
preprocessing to statistical analysis and reproducible reporting, with
a focus on examples from human gut microbiota studies. You
will become familiar with standard bioinformatics concepts and methods
in taxonomic profiling studies of the human microbiome. This includes
better understanding of the specific statistical challenges, practical
hands-on experience with the commonly used methods, and reproducible
research with R.

After the course you will know how to approach new tasks in microbiome
data science by utilizing available documentation and R tools.

**Target audience** Advanced students and applied researchers who wish
  to develop their skills in microbial community analysis.

**Venue** [Radboud University / Online](), Nijmegen. July 5-16, 2021,
  with contributions by [University of
  Turku](http://datascience.utu.fi), Finland.





## Acknowledgments

**Citation** "Introduction to miaverse (2021). Tuomas Borman, Felix Ernst, Sudarshan Shetty, Henrik Eckermann, Leo Lahti. URL: https://microbiome.github.io".

**Contact**

 * [Leo Lahti](http://datascience.utu.fi), University of Turku
 * [miaverse collective](https://microbiome.github.io)

**License** All material is released under the open [CC BY-NC-SA 3.0 License](LICENSE).

**Source code**

The source code of this repository is fully reproducible and contains
the Rmd files with executable code. All files can be rendered at one
go by running the file [main.R](main.R). You can check the file for
details on how to clone the repository and convert it into a gitbook,
although this is not necessary for the training.

- Source code (github): [miaverse teaching material](https://github.com/microbiome/course_2021_radboud)
- Course page (html): [miaverse teaching material](https://microbiome.github.io/course_2021_radboud/)


